basis_set_exchange/manip.py

```diff
@@ -514,29 +522,51 @@ def extend_dunning_aug(basis, level, use_copy=True, as_component=False):
 
             if len(sorted_exponents) < 2:
                 raise RuntimeError(
-                    "Need more than two exponents to extend dunning augmentation. Element {} has {}: ".format(
+                    "Need at least two exponents to for geometric augmentation. Element {} has {}: ".format(
                         el_sym, ','.join(shell['exponents'])))
 
-            alpha, alpha_idx = sorted_exponents[0]
-            beta_tmp, _ = sorted_exponents[1]
-            beta = alpha / beta_tmp
-
-            if alpha == beta_tmp:
+            if steep:
+                # If we're augmenting by steep functions, the
+                # references are the steepest and second-steepest
+                # function.
+                ref_idx=-1
+                next_idx=-2
+            else:
+                # If we're augmenting by diffuse functions, the
+                # references are the diffusemost and second-most
+                # diffuse function.
+                ref_idx=0
+                next_idx=1
+
+            ref_exp, ref_idx = sorted_exponents[ref_idx]
+            next_exp, next_idx = sorted_exponents[next_idx]
+            # Even-tempered spacing parameter
+            beta = ref_exp / next_exp
+
+            if ref_exp == next_exp:
                 raise RuntimeError(
-                    "Two smallest exponents are the same. Duplicate exponents are not a good thing here. Exponent: {}".
-                    format(alpha))
-
-            # Test that the primitive for alpha is completely uncontracted
-            alpha_coefs = [float(c[alpha_idx]) for c in shell['coefficients']]
-            n_coefs = len(alpha_coefs)
-            if alpha_coefs.count(0.0) != n_coefs - 1 or alpha_coefs.count(1.0) != 1:
```

Comment:
So I'm wondering about this piece of code that I adopted for use below. It seems to me that this only applies to optimized general contractions, right? A free primitive can also exist as a free primitive...